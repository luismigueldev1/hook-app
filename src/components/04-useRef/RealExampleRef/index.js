import React, { useState } from "react";
import MultipleCustomHooks from "../../03-examples/MultipleCustomHooks";

export default function RealExampleRef() {
  const [show, setShow] = useState(false);

  const handleShow = () => setShow(!show);
  return (
    <div>
      <h1>Real Example useRef</h1>
      <hr />
      {show && <MultipleCustomHooks />}

      <button className="btn btn-warning mt-2" onClick={handleShow}>
        Show/Hide
      </button>
    </div>
  );
}
