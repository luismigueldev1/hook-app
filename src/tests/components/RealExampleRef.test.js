import { shallow } from "enzyme";
import RealExampleRef from "../../components/04-useRef/RealExampleRef";

describe("Pruebas en <RealExampleRef/>", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("Debe rederizarse correctamente", () => {});

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);

  test("Debe rederizarse el componente MultipleCustomHooks", () => {
    expect(wrapper.find("button").simulate("click"));
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
