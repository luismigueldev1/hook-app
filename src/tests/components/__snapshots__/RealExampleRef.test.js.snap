// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-warning mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
